Sources/SwiftDoubleRatchet/SwiftDoubleRatchet.swift

```diff
@@ -123,34 +113,34 @@ public struct Ratchet<C: Curve> {
         header: Header, ciphertext: C, ad: AD
     ) throws -> Data? {
         let publicKeyNumber = PublicKeyNumber(p: header.publicKey, n: UInt(header.Ns))
-        if case let Optional.some(mk) = Mkskipped.removeValue(forKey: publicKeyNumber) {
-            var concated = Data.init()
-            concated.append(contentsOf: .init(ad))
-            concated.append(header.encoded)
-            return try decrypt(mk: mk, ciphertext: ciphertext, ad: concated)
-        } else {
+        guard let mk = Mkskipped.removeValue(forKey: publicKeyNumber) else {
             return nil
         }
+        
+        var concated = Data.init()
+        concated.append(contentsOf: .init(ad))
+        concated.append(header.encoded)
+        return try decrypt(mk: mk, ciphertext: ciphertext, ad: concated)
     }
     
     private mutating func skipMessageKeys(until: UInt) throws {
-        if Nr + MaxSkip < until {
+        guard Nr + MaxSkip >= until else {
             throw RatchetError.skippedToMany
         }
         
-        if case Optional.some(_) = CKr {
-            if case let Optional.some(curDHr) = DHr {
-                let curDHrData = curDHr.rawRepresentation
-                var mk: SymmetricKey? = nil
-                while Nr < until {
-                    (CKr, mk) = kdfCk(ck: CKr!)
-                    let key = PublicKeyNumber(p: curDHrData, n: Nr)
-                    Mkskipped[key] = mk!
-                    Nr += 1
-                }
-            } else {
+        if CKr != nil {
+            guard let curDHr = DHr else {
                 throw RatchetError.noRecieveReceiveChain
             }
+            
+            let curDHrData = curDHr.rawRepresentation
+            var mk: SymmetricKey
```

Comment:
This doesn't have to be an optional here, because you are always assigning it a value in the loop before it is used.